src/main/java/smile/identity/core/keys/SignatureKey.java

```diff
@@ -0,0 +1,49 @@
+package smile.identity.core.keys;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.charset.StandardCharsets;
+import java.security.GeneralSecurityException;
+import java.time.Instant;
+import java.util.Base64;
+
+public class SignatureKey {
+
+    private Long timestamp;
+    private String signature;
+
+    public SignatureKey(Long timestamp, String partnerId, String apiKey) {
+        this.timestamp = timestamp;
+        generateKey(partnerId, apiKey);
+    }
+
+    private void generateKey(String partnerId, String apiKey) {
```

Comment:
Nit but stylewise would prefer if this method returned the signature and it was set against the object in the constructor rather than a void method.